packages/sfui/frameworks/react/hooks/usePagination/usePagination.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-return-assign */
+import { useState, useEffect } from 'react';
+import paginate from 'jw-paginate';
+
+interface UsePaginationParameters {
+  totalItems: number;
+  currentPage?: number;
+  pageSize?: number;
+  maxPages?: number;
+}
+
+export function usePagination({ totalItems, currentPage = 1, pageSize = 10, maxPages = 1 }: UsePaginationParameters) {
+  const [pagination, setPagination] = useState(paginate(totalItems, currentPage, pageSize, maxPages));
+  const [selectedPage, setSelectedPage] = useState(currentPage);
+
+  useEffect(() => {
+    setSelectedPage(currentPage);
+  }, [currentPage]);
```

Comment:
Question: Is it not needed to write `pagination.currentPage = currentPage` here?